### PR TITLE
[libc++][spaceship] Marks P1614 as complete.

### LINKED ISSUE #100018: Complete 'P1614R2 The Mothership has Landed'
**Link:** https://wg21.link/p1614r2

I think this is currently tracked by https://libcxx.llvm.org/Status/Spaceship.html. We should consider creating a project or milestone for the remaining tasks.

### PR DESCRIPTION
**Pull request by mordante**
Implements parts of:
- P1902R1 Missing feature-test macros 2017-2019

Completes:
- P1614R2 The Mothership has Landed

Fixes #100018

### CODE REVIEW

File: libcxx/docs/Status/SpaceshipPapers.csv [hunk 1]
```diff
@@ -1,5 +1,5 @@
 "Number","Name","Status","First released version"
```
Comment (ldionne) [inline libcxx/docs/Status/SpaceshipPapers.csv:1]: Could we actually get rid of this file? I don't think it serves much of a purpose anymore. We should make sure that all the information in that file is tracked in our other `Cxx20Papers.rst` & friends and then get rid of it.

Comment (mordante) [reply libcxx/docs/Status/SpaceshipPapers.csv:1]: This page has been shipped in several libc++ releases. I strongly like to have it shipped in one release where P1614 is finished. I already have a TODO on my private list to remove it after branching LLVM 20.

**Comment by h-vetinari** [comment]
This now closes #100018

**Review by ldionne — APPROVED** [review]
LGTM. Let's cherry-pick.

**Comment by mordante** [comment]
/cherry-pick [3ce6f68](https://github.com/llvm/llvm-project/commit/3ce6f68ee736e2e6afaa2fe5fcde71f26b21d0cf)

**Pull request merged**